p2p/host/relay/autorelay.go

```diff
@@ -84,6 +84,12 @@ func (ar *AutoRelay) background(ctx context.Context) {
 	// when true, we need to identify push
 	push := false
 
+	reachabilityChanged := make(chan struct{}, 1)
+	defer close(reachabilityChanged)
```

Comment:
This will cause the `findRelays` loop to spin till it sees that the context has been canceled.